lang/rust/avro/src/codec.rs

```diff
@@ -183,6 +197,72 @@ impl Codec {
     }
 }
 
+#[cfg(feature = "bzip")]
+pub mod bzip {
+    use bzip2::Compression;
+
+    #[derive(Clone, Copy, Eq, PartialEq, Debug)]
+    pub struct Bzip2Settings {
```

Comment:
I wonder: do we need **Settings for each codec ?
All 3 classes are the same now.
If we need to add custom codec specific settings later it should not be a big deal - the other codecs will just ignore any setting they don't support. I do not expect many new settings.

Reply:
This won't work because the codecs use different compression level by default and if more than one feature is enabled then it is not possible to decide what to use as a default.